### PR TITLE
fix(credential-eip712): chainId for any did method

### PR DESCRIPTION
**Pull request by simonas-notcat**
## What issue is this PR fixing

Getting correct chainId for any did method

## Quality
Check all that apply:
* [X] I want these changes to be integrated
* [X] I successfully ran `pnpm i`, `pnpm build`, `pnpm test`, `pnpm test:browser` locally.
* [X] I allow my PR to be updated by the reviewers (to speed up the review process).
* [X] I added unit tests.
* [ ] I added integration tests.
* [ ] I did not add automated tests because _________, and I am aware that a PR without tests will likely get rejected.

### CODE REVIEW

File: packages/utils/src/did-utils.ts [hunk 1]
```diff
@@ -161,7 +161,7 @@ export function getEthereumAddress(verificationMethod: VerificationMethod): stri
  *
  * @beta This API may change without a BREAKING CHANGE notice.
  */
-export function getChainIdForDidEthr(verificationMethod: _NormalizedVerificationMethod): number {
+export function getChainId(verificationMethod: _NormalizedVerificationMethod): number {
   if (verificationMethod.blockchainAccountId?.includes('@eip155')) {
```
Comment (stanleyyconsensys) [inline packages/utils/src/did-utils.ts:165]: may be we can also add is NAN as the parseInt result 

if it is nan, then throw error, hence the Eip712 fallback to use chainId 1?

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]
LGTM, leave some comments

File: packages/utils/src/did-utils.ts (hunk 1 above)
Comment (mirceanis) [reply packages/utils/src/did-utils.ts:165]: good point

**Review by mirceanis — APPROVED** [review]
Looks good.
It is worth handling the NaN case that @stanleyyconsensys mentioned.

The test would be against a malformed blockchainAccountId:
eip155:0xe708:0x197...
eip155:linea:0x197...
0x197...@eip155:0xe708
0x197...@eip155:linea

**Review by mirceanis — APPROVED** [review]
Looks good

**Pull request merged**